### PR TITLE
Implementowanie route api

### PR DESCRIPTION
**Pull request by JanBezler**

### CODE REVIEW

File: TripPlanner - Project/APIsApp/views.py [hunk 1]
```diff
@@ -1,8 +1,32 @@
...
     queryset = Route.objects.all()
-    serializer_class = RouteSerializer
\ No newline at end of file
+    serializer_class = RouteSerializer
+
+def buttonSiteView(request):
+    
+    if request.method == "POST":
+        form = CoordinatesForm(request.POST)
+        if form.is_valid():
```
Comment (marcinbar7) [inline TripPlanner - Project/APIsApp/views.py]: Let's say `form.is_valid()` will return `False`. What will happen then?

File: TripPlanner - Project/APIsApp/views.py [hunk 2]
```diff
@@ -1,8 +1,32 @@
...
+from .routeapi import RouteApiRequest
 
 
 class RouteViewSet(viewsets.ModelViewSet):
     queryset = Route.objects.all()
-    serializer_class = RouteSerializer
\ No newline at end of file
+    serializer_class = RouteSerializer
+
+def buttonSiteView(request):
```
Comment (marcinbar7) [inline TripPlanner - Project/APIsApp/views.py]: Please use snake case for all functions and methods as we do in Python.

File: TripPlanner - Project/APIsApp/views.py [hunk 3]
```diff
@@ -1,8 +1,32 @@
...
+            latitude_start = form.cleaned_data['latitude_start']
+            longitude_end = form.cleaned_data['longitude_end']
+            latitude_end = form.cleaned_data['latitude_end']
+
+            route = RouteApiRequest(longitude_start,latitude_start,longitude_end,latitude_end)
+
+            route_model = Route(longitude_start=longitude_start,latitude_start=latitude_start,longitude_end=longitude_end,latitude_end=latitude_end,distance=route.give_distance(),coordinates_json=route.give_coordinates())
+            route_model.save()
+
+            return redirect("/api/button")
```
Comment (marcinbar7) [inline TripPlanner - Project/APIsApp/views.py]: Please, be consistent in the project and don't mix `'` with `"` (applies for all strings). I suggest using single quotes.

File: TripPlanner - Project/APIsApp/views.py [hunk 4]
```diff
@@ -1,8 +1,32 @@
...
+        if form.is_valid():
+
+            longitude_start = form.cleaned_data['longitude_start']
+            latitude_start = form.cleaned_data['latitude_start']
+            longitude_end = form.cleaned_data['longitude_end']
+            latitude_end = form.cleaned_data['latitude_end']
+
+            route = RouteApiRequest(longitude_start,latitude_start,longitude_end,latitude_end)
+
+            route_model = Route(longitude_start=longitude_start,latitude_start=latitude_start,longitude_end=longitude_end,latitude_end=latitude_end,distance=route.give_distance(),coordinates_json=route.give_coordinates())
```
Comment (marcinbar7) [inline TripPlanner - Project/APIsApp/views.py]: Maybe we should use `black formatter` in this project to automate styling.
This line is a bit long. Maybe each line for key=val would a good idea.

File: TripPlanner - Project/APIsApp/views.py [hunk 5]
```diff
@@ -1,8 +1,32 @@
...
+            route_model.save()
+
+            return redirect("/api/button")
+    else:
+        form = CoordinatesForm()
+        return render(request, "button.html",{"form":form})
```
Comment (marcinbar7) [inline TripPlanner - Project/APIsApp/views.py]: Please, remember to add an empty line at the end of the file.

File: TripPlanner - Project/APIsApp/views.py [hunk 6]
```diff
@@ -1,8 +1,32 @@
+from django.shortcuts import render, redirect
 from rest_framework import viewsets
 from APIsApp.serializers import RouteSerializer
 from APIsApp.models import Route
+from APIsApp.forms import CoordinatesForm
+from .routeapi import RouteApiRequest
```
Comment (marcinbar7) [inline TripPlanner - Project/APIsApp/views.py]: Perhaps we could use some tool to sort our import (like `isort-black-compat`)

**Review by marcinbar7 — COMMENTED** [review]
Good job!
To make it a full-fledged api we will have to make some adjustments, but it's a good starting point.

File: TripPlanner - Project/APIsApp/views.py (hunk 1 above)
Comment (marcinbar7) [reply TripPlanner - Project/APIsApp/views.py]: I mean, we probably will have to also handle this case.

**Review by marcinbar7 — COMMENTED** [review]
Looks good to me.
Could you add unit test?

**Comment by ochmanski** [comment]
Really nice!

Could you maybe try using imperative mood in your git commit messages?

Examples:
`Added possibility to make route by POST method` -> `Add possibility to make route by POST method`
`Removed forms.py` -> `Remove forms.py`
I think you get the idea.

It's better to learn it and stick with it now than later when you've developed your habits.

> Write your commit message in the imperative: "Fix bug" and not "Fixed bug"
or "Fixes bug."  This convention matches up with commit messages generated
by commands like git merge and git revert.

More on that here
https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html

**Comment by JanBezler** [comment]
> Really nice!
> 
> Could you maybe try using imperative mood in your git commit messages?
> 
> Examples:
> `Added possibility to make route by POST method` -> `Add possibility to make route by POST method`
> `Removed forms.py` -> `Remove forms.py`
> I think you get the idea.
> 
> It's better to learn it and stick with it now than later when you've developed your habits.
> 
> > Write your commit message in the imperative: "Fix bug" and not "Fixed bug"
> > or "Fixes bug."  This convention matches up with commit messages generated
> > by commands like git merge and git revert.
> 
> More on that here
> https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html

Oh okay, honestly I have never heard  about that before

File: TripPlanner - Project/APIsApp/test_routeapi.py [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+error_response = {"error": {"code": 2004, "message": "Request parameters exceed the server configuration limits. The approximated route distance must not be greater than 6000000.0 meters."}, "info": {
+    "engine": {"version": "6.3.6", "build_date": "2021-02-21T01:31:06Z"}, "timestamp": 1614100819473}}
+
+good_object = RouteApiRequest(8.681495, 49.41461, 8.687872, 49.420318)
+bad_object = RouteApiRequest(200, 200, 200, 200)
```
Comment (konecka-p) [inline TripPlanner - Project/APIsApp/test_routeapi.py]: It may be worth using pytest's fixtures to get and set up test data.

Comment (marcinbar7) [reply TripPlanner - Project/APIsApp/test_routeapi.py]: Right, let's do it in a new ticket, @konecka-p would you like to do it?

**Review by marcinbar7 — APPROVED** [review]

**Pull request merged**